### ISSUE TITLE
hlint suggests avoiding NonEmpty’s head/last/init functions even though they’re safe

### ISSUE THREAD

**Issue by patrickt** [ndmitchell/hlint#787]
When evaluating the code

```haskell
for_ (NonEmpty.init modulePaths) require
```

```
src/Language/Python/Syntax.hs:152:11: Warning: Avoid restricted function
Found:
  NonEmpty.init
Note: may break the code
```

`init` over `[]` is indeed a pitfall, so I commend `hlint` on its paranoia, but here I don’t think it’s called for. I recognize that a general solution requires a running Haskell interpreter, but perhaps this warning can be suppressed if there is an explicit `NonEmpty` qualifier in the function invocation.

**Comment by patrickt** [comment]
Ah, I think this was my error: I had a clause earlier to warn on `init` etc. that I had forgotten about. Sorry for the noise!
